delivery_carrier_label_postlogistics/postlogistics/web_service.py

```diff
@@ -330,40 +336,52 @@ def _prepare_data(self, lang, frankingLicense, post_customer,
         }
 
     def get_access_token(self, env):
-        icp = env['ir.config_parameter'].sudo()
-        client_id = icp.get_param('postlogistics.oauth.client_id')
-        client_secret = icp.get_param('postlogistics.oauth.client_secret')
-        authentication_url = icp.get_param(
-            'postlogistics.oauth.authentication_url'
-        )
+        """Threadsafe access to token"""
+
+        with self._lock:
+            now = datetime.now()
+
+            if self.access_token:
+                # keep a safe margin on the expiration
+                expiry = self.access_token_expiry - timedelta(seconds=5)
+                if now < expiry:
+                    return
```

Comment:
In case of you have a token not expired, I think return True will not work, because later in the file you take the token as returned value of `get_access_token`:
https://github.com/OCA/delivery-carrier/blob/12.0/delivery_carrier_label_postlogistics/postlogistics/web_service.py#L387

Reply:
rewritten to actually write on the class.

It's ok to have each worker creating a new class and having a separate access_token in parallel.